### PR TITLE
[16.0][ADD] purchase_order_restriction

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
The purpose of this module is to limit the access to the purchase order to 1) the buyer and 2) purchase managers.

@qrtl

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Isn't this the same as https://github.com/OCA/purchase-workflow/tree/15.0/purchase_security ?

**Comment by AungKoKoLin1997** [comment]
@pedrobaeza These are not the same. `purchase_security` allows only see his own purchase order for Purchase (own orders) group. This module adds new field `is_restricted` in purchase order that can only be changed by its own purchase user or purchase manager. So, if purchase order is restricted by its own purchase or purchase manager, this order can only be seen by its own purchase user or purchase manager.

**Comment by pedrobaeza** [comment]
Ok then

**Review by yostashiro — COMMENTED** [review]
@AungKoKoLin1997 Can you please add some tests.

File: purchase_order_restriction/tests/test_purchase_order_restriction.py
Comment (yostashiro) [inline purchase_order_restriction/tests/test_purchase_order_restriction.py:1]: @AungKoKoLin1997 Can you please check if we can inherit the test class from the purchase module.

Comment (AungKoKoLin1997) [reply purchase_order_restriction/tests/test_purchase_order_restriction.py:1]: @yostashiro  There is no much effective inheriting test class of purchase module. We can only use vendor value if we inherit.

File: purchase_order_restriction/tests/test_purchase_order_restriction.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    def test_order_restriction(self):
+        # PO user should have access to all of them
+        # before it's restricted
+        self.assertEqual(
+            len(
+                self.env["purchase.order"]
+                .with_user(self.user1)
+                .search([("name", "like", "Purchaser Order")])
+                .ids
```
Comment (yostashiro) [inline purchase_order_restriction/tests/test_purchase_order_restriction.py]:
```suggestion
```

File: purchase_order_restriction/tests/test_purchase_order_restriction.py
Comment (yostashiro) [reply purchase_order_restriction/tests/test_purchase_order_restriction.py:1]: @AungKoKoLin1997 Then can you try and reduce LOC by creating methods such as _create_user(), _create_order(), etc.?

File: purchase_order_restriction/tests/test_purchase_order_restriction.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    def _create_user(self, name, login, email, group):
+        return (
+            self.env["res.users"]
+            .with_context(no_reset_password=True)
+            .create(
+                {
+                    "name": name,
+                    "login": login,
+                    "email": email,
```
Comment (yostashiro) [inline purchase_order_restriction/tests/test_purchase_order_restriction.py]: `email` can be fixed as it doesn't matter.

```suggestion
                    "email": "test@yourcompany.com",
```

**Review by yostashiro — APPROVED** [review]
Code review and functional test. 👍 

Please squash commits.

**Review by navarrorico — APPROVED** [review]
Functional review

**Comment by AungKoKoLin1997** [comment]
@dreispt The merge was failed. Can you please try again?

**Comment by HviorForgeFlow** [comment]
@AungKoKoLin1997 merge will still failing because purchase_order_restriction is not compatible with purchase_security tests.

It will be necessary to create new tests ci entry excluding one each other.

@pedrobaeza any hint?

**Comment by pedrobaeza** [comment]
Why not adding the feature `is_restricted` to `purchase_security` instead?

**Pull request closed without merging**